tests/functional/behave_features/common/utils/owner_file_submission.py

```diff
@@ -0,0 +1,351 @@
+            self.temp_dir.cleanup()
+
+        if self.repo_manager:
+            self.repo_manager.repo.git.worktree("prune")
+
+    def submission_path(self) -> str:  # charts/partner/mycompany/mychartname
+        """Composes the submission path based on the vendor_type, vendor, and chart_name values."""
+        return os.path.join(
+            "charts", self.vendor_category, self.vendor_label, self.chart_name
+        )
+
+    def set_git_username_email(self, username: str, email: str):
+        """Writes the username and email to the repo's .git/config.
+
+        Note that calling this will overwrite repository-local values, which
+        will supercede global values.
+
+        Args:
+            repo: git.Repo instance of the local directory
+            username: git username to set
+            email: git email to set
+        """
+        self.repo_manager.repo.config_writer().set_value(
+            "user", "name", username
+        ).release()
+        self.repo_manager.repo.config_writer().set_value(
+            "user", "email", email
+        ).release()
+
+    def create_and_commit_owners_file(self):
+        self.repo_manager.checkout_branch(self.secrets.base_branch)
+        self.repo_manager.push_branch(self.secrets.test_repo, self.secrets.base_branch)
+        self.repo_manager.checkout_branch(self.secrets.pr_branch)
+        return self._create_and_commit_owners_file(
+            self.submission_path(),
+            self.secrets.pr_branch,
+            self.chart_name,
+            self.vendor_label,
+            self.vendor_name,
+        )
+
+    def _create_and_commit_owners_file(
+        self,
+        chart_directory: str,
+        pr_branch: str,
+        chart_name: str,
+        vendor_label: str,
+        vendor_name: str,
+    ):
+        """Creates an OWNERS file in the PR branch."""
+        with SetDirectory(Path(self.temp_dir.name)):
+            # Create the OWNERS file from the string template
+            values = {
+                "bot_name": self.secrets.bot_name,
+                "vendor": vendor_label,
+                "vendor_pretty": vendor_name,
+                "chart_name": chart_name,
+                "public_key": "null",
+                "provider_delivery": "false",
+            }
+            content = Template(base_owners_file).substitute(values)
+            logging.debug(f"OWNERS File Content:\n{content}")
+            pathlib.Path(chart_directory).mkdir(parents=True, exist_ok=True)
+            with open(f"{chart_directory}/OWNERS", "w+") as fd:
+                fd.write(content)
+
+            logging.info(f"Push OWNERS file to '{self.secrets.test_repo}:{pr_branch}'")
+            worktree_repo = git.Repo()
+            worktree_repo.git.add(f"{chart_directory}/OWNERS")
+            worktree_repo.git.commit("-m", f"Add {chart_name} OWNERS file")
+            self.repo_manager.push_branch(
+                self.secrets.test_repo, pr_branch, repo=worktree_repo
+            )
```

Comment:
Not technically. One of the things this (and the chart cert) workflow tries to do is confine "filesystem modifications" to worktrees so that the base repo state doesn't change underneath the CI.

When I created repo_manager at the instantiation of this class, it references the original worktree, and not any additional worktree, regardless of the current working directory. As such, I create a Repo instance referencing the worktree directory, which is where I'm commit changes. That allows me to "git push" the branch to which this worktree refers without having to change the underlying base repo directory.

I suppose I could checkout the branch containing changes, but again, we start to muck around with the filesystem path that's under CI at any given time. Referring to the worktree seems easier.